oc-chef-pedant/lib/pedant/rspec/search_util.rb

```diff
@@ -718,7 +718,9 @@ def direct_solr_query(type, query)
       end
 
       def get_response_count(r)
-        if r["response"].nil?
+        if (r["response"].nil?) and (r["hits"]["total"].is_a? Hash)
+          r["hits"]["total"]["value"]
```

Comment:
Is this trying to fix for the case when there is data in the system?

Reply:
get_response_count function is to return the number documents found.There is a response structure change for  total field in the ES7.
`"hits" : {
    "total" : {
      "value" : 1,
      "relation" : "eq"
    },
    "max_score" : 1.0,
    "hits" : [
      {
        "_index" : "chef",
        "_type" : "_doc",
        "_id" : "2bed8b78e2e51e0c13c82bbbf07f0ad1",
        "_score" : 1.0
      }]}`

in previous versions, Total filed is directly a number indicating the number of documents found.So Added another simple condition to check whether it is a another json or not.

Reply:
With the `and`, I think parentheses aren't necessary.  But if you had used `&&`, they would be necessary:
```
irb(main):061:0> true and {}.is_a? Hash
=> true
irb(main):062:0> false and {}.is_a? Hash
=> false
irb(main):063:0> false or {}.is_a? Hash
=> true
irb(main):064:0> true && {}.is_a? Hash
Traceback (most recent call last):
        1: from C:/Ruby25-x64/bin/irb.cmd:19:in `<main>'
SyntaxError ((irb):64: syntax error, unexpected tCONSTANT, expecting end-of-input)
true && {}.is_a? Hash
                 ^~~~
irb(main):065:0> true && ({}.is_a? Hash)
=> true
```
Up to you if you consider parentheses more readable.